connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginUtils.java

```diff
@@ -325,6 +328,34 @@ public static List<Path> pluginUrls(Path topPath) throws IOException {
         return Arrays.asList(archives.toArray(new Path[0]));
     }
 
+    public static Set<PluginSource> pluginSources(List<Path> pluginLocations, DelegatingClassLoader classLoader, ClassLoaderFactory factory) {
```

Comment:
Any reason not to generalize the signature, since we don't use any methods specific to the `DelegatingClassLoader` class?
```suggestion
    public static Set<PluginSource> pluginSources(List<Path> pluginLocations, ClassLoader classLoader, ClassLoaderFactory factory) {
```